mobile/lib/api/call.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:triggo/api/codes.dart';
+import 'package:triggo/api/response.dart';
+import 'package:triggo/constants/config/api.dart';
+
+Future<CallResponse<R>?> call<T, R>({
+  required String method,
+  required String endpoint,
+  T? body,
+  String? bearerToken,
+  http.Client? client,
+}) async {
+  final http.Client httpClient = client ?? http.Client();
+
+  try {
+    final uri = Uri.parse('${Config.apiUrl}$endpoint');
+    final headers = {
+      'Content-Type': 'application/json',
+      if (bearerToken != null) 'Authorization': 'Bearer $bearerToken',
+    };
+
+    http.Response response;
+    switch (method.toUpperCase()) {
+      case 'POST':
+        response = await httpClient.post(
+          uri,
+          headers: headers,
+          body: body != null ? jsonEncode(body) : null,
+        );
+        break;
+      case 'PUT':
+        response = await httpClient.put(
+          uri,
+          headers: headers,
+          body: body != null ? jsonEncode(body) : null,
+        );
+        break;
+      case 'DELETE':
+        response = await httpClient.delete(
+          uri,
+          headers: headers,
+          body: body != null ? jsonEncode(body) : null,
+        );
+        break;
+      case 'GET':
+        response = await httpClient.get(uri, headers: headers);
+        break;
+      default:
+        throw Exception('Invalid method');
+    }
```

Comment:
Dear Colleague, do you know if you can replace this switch by a thing like an array of function pointers, or something like that in flutter ? Matheo told me to check near the `yield` function

Reply:
Oh! ur right! `yield` enable pattern matching, I'll do that. Thx

Reply:
Never mind, I do some research about the keyword `yield` and it doesn't really apply to this function. yield is used in a async context with several steps. For example, if we have a downloading feature, we'll use this because we can notify the front about our downloading percentage. IDK if you get it ?

Reply:
Yes, I get it, and did you find something about the possibility to replace this switch ?

Reply:
Hmmm, not really. I found other thing like the `sealed` class who will enable us pattern matching but it might be too verbose for this little function

Reply:
But I even updated my code. It should be more readable rn.